### ISSUE TITLE
Avoid using `build/showMessage` for ZincWorker cache.

### ISSUE THREAD

**Issue by ckipp01** [com-lihaoyi/mill#1595]
When using mill as a BSP server you'll occasionally see a message saying `ZinkWorker: using cache size 1` right in your editor. I don't think this is a great message to surface to the user since it's not actionable and arguable some users will have no idea what it means. This is sent from mill via:

```json
[Trace - 02:21:27 PM] Received notification 'build/showMessage'
Params: {
  "type": 4,
  "message": "ZinkWorker: using cache size 1"
}
```
Looking through the code the only place I see this is:

https://github.com/com-lihaoyi/mill/blob/c3b64a54be184d9d41b219d84761479dd61f10af/scalalib/src/ZincWorkerModule.scala#L52

But it's a debug  message, so I'm confused how this is floating up.  If I understand it I see the logger in here:

https://github.com/com-lihaoyi/mill/blob/c3b64a54be184d9d41b219d84761479dd61f10af/bsp/src/mill/bsp/MillBspLogger.scala#L52-L55

This is getting captured here, but again I'm unsure how this is floating up as I don't have debug set. Also, do you think it might be better to use https://build-server-protocol.github.io/docs/specification.html#log-message instead of showMessage for some of the debug messages? I feel like https://build-server-protocol.github.io/docs/specification.html#show-message should be used sparingly and only when there is some type of actionable output to the user or informative information that they'd what to know. What do you think?

**Comment by lefou** [comment]
The interesting thing is how this kind of message should be handled? BSP defined 4 kind of message types: `ERROR`, `WARNING`, `INFORMATION` and `LOG`. This suggests that `LOG` level is below `INFORMATION` which is typically used for debug messages. Although I agree that it is questionable whether we should log this dedicated message at all, we should pass debug logging to the user somehow. Who's responsibility is to enable/disable debug logging. Should the client do it and just not show the message or should the server never send the message?

**Comment by lefou** [comment]
As Mill BSP support is still in heavy development we want to see all message including debug in the `.bsp/mill-bsp.stderr` file. (This files catches everthing that goes to STDERR, for convenience and to not confuse the BSP client.) That's why we currently default (and hardcode I think) to enable debug logging.

**Comment by ckipp01** [comment]
> Who's responsibility is it to enable/disable debug logging. Should the client do it and just not show the message or should the server never send the message?

I believe it can be either. Either a users if they know how can turn on debugging for the server or the client can as well if it needs to. As for the second part, if I understand what you mean, it's not so much about choosing to show it or not, but rather just handling the information in the way that it's given. Since even the debug messages are `build/showMessage` it will directly float up to the user in their UI.

Relating the extra files in `.bsp/*`, just a heads up that `mill-bsp.trace` can really balloon in size. In metals we also have the ability to have lsp and bsp traces files, but don't by default since in the lifecycle of working on a projects these files can become giant.

**Comment by lefou** [comment]
@ckipp01 Thanks for your input. 

I removed that particular message, as it dates back to the times we experimented with parallel support, which is now pretty stable.

More generally, I think we will keep the concept of passing all messages via BSP, regardless if they come from the compiler or some other source in Mill. After all, we try to support much more use-cases with BSP that just Java and Scala via Zinc.

Thanks for the reminder about log file size. I like the idea for the trace file, to just write to it if it's already present.
